std/algorithm/searching.d

```diff
@@ -696,11 +696,16 @@ size_t count(alias pred = "true", R)(R haystack)
 if (isInputRange!R && !isInfinite!R &&
     is(typeof(unaryFun!pred(haystack.front)) : bool))
 {
-    size_t result;
-    alias T = ElementType!R; //For narrow strings forces dchar iteration
-    foreach (T elem; haystack)
-        if (unaryFun!pred(elem)) ++result;
-    return result;
+    static if (hasLength!R && is(typeof(unary!fun(pred) == unary!fun("true"))))
```

Comment:
Your `is` swallows all errors (even your typos), but comparing functions doesn't work in D, e.g.:

```
void main(string[] args)
{
    import std.functional;
    static assert(is(typeof(unaryFun!("true") == unaryFun!("true"))));
}
```

Regarding the specialization, you should see https://github.com/dlang/phobos/pull/4265#issuecomment-223762623 and its final version (https://github.com/dlang/phobos/pull/5001).
tl;dr: use overloads, not strings